### PR TITLE
Deduplicate test command creation

### PR DESCRIPTION
**Pull request by konstin**
This PR refactors the command creation in the test suite to remove the duplication.

**1)** We add the same set of test stubbing args to almost any uv invocation in the tests:

```rust
command
    .arg("--cache-dir")
    .arg(self.cache_dir.path())
    .env("VIRTUAL_ENV", self.venv.as_os_str())
    .env("UV_NO_WRAP", "1")
    .env("HOME", self.home_dir.as_os_str())
    .env("UV_TOOLCHAIN_DIR", "")
    .env("UV_TEST_PYTHON_PATH", &self.python_path())
    .current_dir(self.temp_dir.path());

if cfg!(all(windows, debug_assertions)) {
    // TODO(konstin): Reduce stack usage in debug mode enough that the tests pass with the
    // default windows stack of 1MB
    command.env("UV_STACK_SIZE", (8 * 1024 * 1024).to_string());
}
```

Centralizing these into a `TestContext::add_shared_args` method removes them from everywhere.

**2)** Prefix all `TextContext` methods of the pip interface with `pip_`. This is now necessary due to `uv sync` vs. `uv pip sync`.

**3)** Move command creation in the various test files into dedicated functions or methods to avoid repeating the arguments. Except for error message tests, there should be at most one `Command::new(get_bin())` call per test file. `EXCLUDE_NEWER` is exclusively used in `TestContext`.

---

I'm considering adding a `TestCommand` on top of these changes (in another PR) that holds a reference to the `TextContext`, has `add_shared_args` as a method and uses `Fn(Self) -> Self` instead of `Fn(&mut Self) -> Self` for methods to improved chaining.

### CODE REVIEW

File: crates/uv/tests/pip_check.rs [hunk 1]
```diff
@@ -6,45 +6,15 @@ use assert_fs::fixture::PathChild;
...
-        // TODO(konstin): Reduce stack usage in debug mode enough that the tests pass with the
-        // default windows stack of 1MB
-        command.env("UV_STACK_SIZE", (2 * 1024 * 1024).to_string());
-    }
-
-    command
-}
-
 /// Create a `pip check` command with options shared across scenarios.
 fn check_command(context: &TestContext) -> Command {
```
Comment (zanieb) [inline crates/uv/tests/pip_check.rs:14]: Shouldn't we just have these in the test context itself?

File: crates/uv/tests/common/mod.rs [hunk 2]
```diff
@@ -312,60 +340,33 @@ impl TestContext {
     /// it can result in tests failing when the index state changes. Therefore,
     /// if you use this, there should be some other kind of mitigation in place.
     /// For example, pinning package versions.
-    pub fn install_without_exclude_newer(&self) -> std::process::Command {
-        let mut command = std::process::Command::new(get_bin());
+    pub fn pip_install_without_exclude_newer(&self) -> Command {
```
Comment (zanieb) [inline crates/uv/tests/common/mod.rs:343]: Honestly I don't think we need a dedicated method for this. People should just `unset_env("EXCLUDE_NEWER")` in the relevant tests?

**Review by zanieb — APPROVED** [review]
Thank you! This was on my backlog :)

File: crates/uv/tests/common/mod.rs (hunk 2 above)
Comment (konstin) [reply crates/uv/tests/common/mod.rs:343]: https://github.com/astral-sh/uv/issues/4531

File: crates/uv/tests/pip_check.rs (hunk 1 above)
Comment (konstin) [reply crates/uv/tests/pip_check.rs:14]: I feel they are fine, feel free to move them though

**Pull request merged**